include/openthread/coap.h

```diff
@@ -455,14 +455,26 @@ const otCoapOption *otCoapHeaderGetNextOption(otCoapHeader *aHeader);
 /**
  * This function creates a new message with a CoAP header.
  *
- * @param[in]  aInstance     A pointer to an OpenThread instance.
- * @param[in]  aHeader  A pointer to a CoAP header that is used to create the message.
+ * @param[in]  aInstance  A pointer to an OpenThread instance.
+ * @param[in]  aHeader    A pointer to a CoAP header that is used to create the message.
  *
  * @returns A pointer to the message or NULL if failed to allocate message.
  *
  */
 otMessage *otCoapNewMessage(otInstance *aInstance, const otCoapHeader *aHeader);
 
+/**
+ * This function creates a new message with a CoAP header and specified priority.
+ *
+ * @param[in]  aInstance  A pointer to an OpenThread instance.
+ * @param[in]  aHeader    A pointer to a CoAP header that is used to create the message.
+ * @param[in]  aPriority  The message priority level.
+ *
+ * @returns A pointer to the message or NULL if failed to allocate message.
+ *
+ */
+otMessage *otCoapNewMessage2(otInstance *aInstance, const otCoapHeader *aHeader, otMessagePriority aPriority);
```

Comment:
Is there any alternate name to `NewMessage2` API?

Perhaps just have the single, original `NewMessage` API that always returns `DefaultPriority` and add a `SetPriority` API for instances when the user wants to use a different priority?

Reply:
What about the name `otCoapNewMessageWithPriority()`?

I also want to use the same API, but there is a difference between the API "NewMessage()" and "NewMessage2()". When there is no memory for the new message, `NewMessage()` will return `NULL`, then you can't call `SetPriority()` to set the priority any more.  But for the function `NewMessage2()`, it will free low priority messages to get memory for the new message.